pkg/model/job.go

```diff
@@ -232,9 +233,17 @@ type Spec struct {
 	Deal Deal `json:"Deal,omitempty"`
 }
 
+var (
+	// The job should automatically receive the longest possible timeout.
+	MaxAllowedJobTimeout time.Duration = time.Duration(0)
+
+	// The job should not be subject to timeouts.
+	NoJobTimeout time.Duration = time.Duration(math.MaxInt64).Truncate(time.Second)
+)
+
```

Comment:
1. Sure thing.
2. Yes, perhaps this is poorly named. The zero value for the timeout has been used for a while as "pick whatever the largest timeout is that I can have" - basically, the user did not supply a timeout. Would you be more happy with:
    - A better name, e.g. `AutomaticTimeout` or sometihng.
    - Turning the timeout field into a pointer and if it's missing apply the default logic. I think this would be backwards compatible in JSON world.